src/QueryManager.ts

```diff
@@ -0,0 +1,222 @@
+import Store from '@orbit/store'
+
+import {
+  Term,
+  OngoingQueries,
+  RecordObject,
+  Expressions,
+  Queries,
+  Subscriptions,
+  QueryRefs,
+  Statuses,
+  QueryCacheOptions,
+} from './types'
+import { Transform, RecordOperation, RecordIdentity } from '@orbit/data'
+import { identityIsEqual } from './helpers'
+
+export class QueryManager<E extends { [key: string]: any } = any>  {
+  _extensions: E
+  _store: Store
+  _ongoingQueries: OngoingQueries
+
+  subscriptions: Subscriptions
+  statuses: Statuses
+
+  constructor (orbitStore: Store, extensions?: E) {
+    this._extensions = extensions || {} as E
+    this._store = orbitStore
+    this._ongoingQueries = {}
+
+    this.subscriptions = {}
+    this.statuses = {}
+  }
+
+  generateQueryRef (queries: Queries) {
+    const terms = this._extractTerms(queries)
+
+    let queryRef = JSON.stringify(terms)
+
+    if (this.subscriptions[queryRef]) return queryRef
+
+    this.subscriptions[queryRef] = { listeners: 0, terms }
+    return queryRef
+  }
+
+  _extractTerms (queries: Queries): Term[] {
```

Comment:
make explicit with `private _extractTerms ...` 
http://www.typescriptlang.org/docs/handbook/classes.html#public-private-and-protected-modifiers